pkg/client/kubernetes/client.go

```diff
@@ -128,9 +129,31 @@ func NewClientFromSecretObject(secret *corev1.Secret, fns ...ConfigFunc) (Interf
 	return nil, errors.New("the secret does not contain a field with name 'kubeconfig'")
 }
 
-// RESTConfigFromClientConnectionConfiguration creates a *rest.Config from a componentbaseconfig.ClientConnectionConfiguration and the configured kubeconfig.
+// RESTConfigFromInternalClientConnectionConfiguration creates a *rest.Config from a componentbaseconfig.ClientConnectionConfiguration and the configured kubeconfig.
 // Allowed fields are not considered unsupported if used in the kubeconfig.
-func RESTConfigFromClientConnectionConfiguration(cfg *componentbaseconfig.ClientConnectionConfiguration, kubeconfig []byte, allowedFields ...string) (*rest.Config, error) {
+// Deprecated: use RESTConfigFromClientConnectionConfiguration instead.
+// TODO(timebertt): delete this when finalizing https://github.com/gardener/gardener/issues/11043
+func RESTConfigFromInternalClientConnectionConfiguration(cfg *componentbaseconfig.ClientConnectionConfiguration, kubeconfig []byte, allowedFields ...string) (*rest.Config, error) {
+	return RESTConfigFromClientConnectionConfiguration(ConvertClientConnectionConfigurationToExternal(cfg), kubeconfig, allowedFields...)
+}
+
+// ConvertClientConnectionConfigurationToExternal converts a componentbaseconfig.ClientConnectionConfiguration to componentbaseconfigv1alpha1.ClientConnectionConfiguration.
+// This function is added for supporting the transition to the external version while dropping the internal version of config APIs.
+// Deprecated: use the external version of ClientConnectionConfiguration directly instead.
+// TODO(timebertt): delete this when finalizing https://github.com/gardener/gardener/issues/11043
+func ConvertClientConnectionConfigurationToExternal(cfg *componentbaseconfig.ClientConnectionConfiguration) *componentbaseconfigv1alpha1.ClientConnectionConfiguration {
+	return &componentbaseconfigv1alpha1.ClientConnectionConfiguration{
+		Kubeconfig:         cfg.Kubeconfig,
+		AcceptContentTypes: cfg.AcceptContentTypes,
+		ContentType:        cfg.ContentType,
+		QPS:                cfg.QPS,
+		Burst:              cfg.Burst,
+	}
+}
+
+// RESTConfigFromClientConnectionConfiguration creates a *rest.Config from a componentbaseconfigv1alpha1.ClientConnectionConfiguration and the configured kubeconfig.
+// Allowed fields are not considered unsupported if used in the kubeconfig.
```

Comment:
Let's get rid of the double negation here
```suggestion
// Allowed fields are considered supported if used in the kubeconfig.
```